expression/builtin_compare_vec.go

```diff
@@ -590,11 +592,113 @@ func (b *builtinNullEQStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	var err error
+	var buf0, buf1 *chunk.Column
+	buf0, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	vecCompareNull(buf0, buf1, result)
+	VecCompareInt(mysql.HasUnsignedFlag(b.args[0].GetType().Flag), mysql.HasUnsignedFlag(b.args[0].GetType().Flag), buf0, buf1, result)
+	vecResOfLT(result)
+	return nil
+}
+
+func vecResOfLT(result *chunk.Column) {
+	res := result.Int64s()
+	n := len(res)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			res[i] = 0
+		}
+		if res[i] < 0 {
+			res[i] = 1
+		} else {
+			res[i] = 0
+		}
+	}
+}
+
+func vecCompareNull(largs, rargs, result *chunk.Column) {
+	res := result.Int64s()
+	n := len(res)
+	for i := 0; i < n; i++ {
+		if largs.IsNull(i) && rargs.IsNull(i) {
+			res[i] = 0
+		} else if largs.IsNull(i) {
+			res[i] = -1
+		} else {
+			res[i] = 1
+		}
+	}
+}
+
+//vectorized CompareInt()
+func VecCompareInt(isUnsigned0, isUnsigned1 bool, largs, rargs, result *chunk.Column) {
+	arg0 := largs.Int64s()
+	arg1 := rargs.Int64s()
+	res := result.Int64s()
+	n := len(arg0)
+
+	switch {
+	case isUnsigned0 && isUnsigned1:
+		for i := 0; i < n; i++ {
+			if result.IsNull(i) {
+				continue
+			}
+			res[i] = int64(types.CompareUint64(uint64(arg0[i]), uint64(arg1[i])))
```

Comment:
1. Maybe we can try to make CompareUint64 and CompareInt64 be vectorized.
2. If we swap line 621 and 622, the check line666-668 and 606-698 can be removed.

Reply:
Can we implement three vectorized functions?
1 VecCompareUU(uint64, uint64)
2 VecCompareII(int64, int64)
3 VecCompareUI(int64, int64)

Reply:
I had implement three vectorized functions.

Reply:
Should the benchmark result in the description be updated?

Reply:
OK, I had updated the benchmark result. Performance is not improved. 
![image](https://user-images.githubusercontent.com/13314548/66894538-409c0880-f023-11e9-86e4-cb2da2708cf9.png)